src/Temporalio/Client/TemporalClient.Workflow.cs

```diff
@@ -472,6 +473,11 @@ private async IAsyncEnumerable<WorkflowExecution> ListWorkflowsInternalAsync(
                             req, rpcOptsAndCancelSource.Item1).ConfigureAwait(false);
                         foreach (var exec in resp.Executions)
                         {
+                            if (input.Options != null && input.Options.Limit > 0 &&
+                                yielded++ >= input.Options.Limit)
```

Comment:
Not the biggest fan of side effects in conditionals like this since postfix increment operator precedence is not always easy for a code reader to understand, but not a blocker.